src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWPApplication.cs

```diff
@@ -394,72 +378,277 @@ internal string LogoPathFromUri(string uri, string theme)
                         { PackageVersion.Windows8, new List<int> { 100 } },
                     };
 
+                if (!highContrast)
+                {
+                    paths.Add(path);
+                }
+
                 if (scaleFactors.ContainsKey(Package.Version))
                 {
                     foreach (var factor in scaleFactors[Package.Version])
                     {
-                        paths.Add($"{prefix}.scale-{factor}{extension}");
-                        paths.Add($"{prefix}.scale-{factor}_{theme}{extension}");
-                        paths.Add($"{prefix}.{theme}_scale-{factor}{extension}");
+                        if (highContrast)
+                        {
+                            paths.Add($"{prefix}.scale-{factor}_{colorscheme}{extension}");
+                            paths.Add($"{prefix}.{colorscheme}_scale-{factor}{extension}");
+                        }
+                        else
+                        {
+                            paths.Add($"{prefix}.scale-{factor}{extension}");
```

Comment:
Why not add colorcheme when it's not in high contrast mode?

Reply:
The function `SetColoredIcon` exactly does that. We call `GetScaleIcons` twice, with `highContrast` values set to true and false. This is to ensure we first check for colored icons and if none are available, use high contrast icons.